nomad/structs/config/tls.go

```diff
@@ -168,3 +182,24 @@ func (t *TLSConfig) Merge(b *TLSConfig) *TLSConfig {
 	}
 	return result
 }
+
+// Equals compares the fields of two TLS configuration objects, returning a
+// boolean indicating if they are the same.
+// NewConfig should never be nil- calling code is responsible for walways
+// passing a valid TLSConfig object
+func (t *TLSConfig) Equals(newConfig *TLSConfig) bool {
+	if t == nil {
+		return false
+	}
+
+	if t == nil && newConfig != nil {
```

Comment:
`t` can't equal `nil` here and newConfig never should be either. Maybe just remove this check?

Reply:
Both could equal nil, although neither _should_. I see your point about removing the checks though, and letting a failure occur in the case the caller does mistakenly does call this method on a nil pointer or pass nil as a parameter.